src/components/Kv/KvMap.vue

```diff
@@ -0,0 +1,297 @@
+	</div>
+</template>
+
+<script>
+import { createIntersectionObserver } from '@/util/observerUtils';
+import testDelayedGlobalLibrary from '@/util/timeoutUtils';
+
+export default {
+	metaInfo() {
+		return {
+			script: [].concat(!this.hasWebGL ? [
+				// leaflet - uses raster tiles for additional browser coverage
+				{
+					vmid: `leafletjs${this.mapId}`,
+					src: 'https://unpkg.com/leaflet@1.7.1/dist/leaflet.js',
+					async: true,
+					defer: true,
+				},
+			] : []).concat(this.hasWebGL ? [
+				// maplibregl - uses vector tiles and webgl for rendering
+				{
+					vmid: `maplibregljs${this.mapId}`,
+					src: 'https://cdn.maptiler.com/maplibre-gl-js/v1.14.0/maplibre-gl.js',
+					async: true,
+					defer: true,
+				},
+			] : []),
+			link: [
+			].concat(!this.hasWebGL ? [
+				// leaflet - uses raster tiles for additional browser coverage
+				{
+					vmid: `leafletcss${this.mapId}`,
+					rel: 'stylesheet',
+					href: 'https://unpkg.com/leaflet@1.7.1/dist/leaflet.css'
+				},
+			] : []).concat(this.hasWebGL ? [
+				// maplibregl - uses vector tiles and webgl for rendering
+				{
+					vmid: `maplibreglcss${this.mapId}`,
+					rel: 'stylesheet',
+					href: 'https://cdn.maptiler.com/maplibre-gl-js/v1.14.0/maplibre-gl.css'
+				},
+			] : []),
+		};
+	},
+	props: {
+		/**
+		 * Aspect Ration for computed map dimensions
+		 * We'll divide the container width by this to determine the height
+		 */
+		aspectRatio: {
+			type: Number,
+			default: 1.3
+		},
+		/**
+		 * Control how quickly the autoZoom occurs
+		 */
+		autoZoomDelay: {
+			type: Number,
+			default: 1500
+		},
+		/**
+		 * These maps require a height to be set in order to render
+		 */
+		height: {
+			type: Number,
+			default: 300,
+		},
+		/**
+		 * Setting this initialZoom will zoom the map from initialZoom to zoom when the map enters the viewport
+		 */
+		initialZoom: {
+			type: Number,
+			default: null
+		},
+		/**
+		 * Set the center point latitude
+		 */
+		lat: {
+			type: Number,
+			default: 37.700091
+		},
+		/**
+		 * Set the center point longitude
+		 */
+		long: {
+			type: Number,
+			default: -123.013243
+		},
+		/**
+		 * Set this if there are more than one map on the page
+		 */
+		mapId: {
+			type: Number,
+			default: 0
+		},
+		/**
+		 * Force use of Leaflet
+		 */
+		useLeaflet: {
+			type: Boolean,
+			default: false
+		},
+		/**
+		 * Default zoom level
+		 */
+		zoomLevel: {
+			type: Number,
+			default: 4
+		},
+	},
+	data() {
+		return {
+			hasWebGL: false,
+			leafletReady: false,
+			mapInstance: null,
+			mapLibreReady: false,
+			mapLoaded: false,
+			zoomActive: false,
+		};
+	},
+	computed: {
+		mapDimensions() {
+			const container = this.$el?.getBoundingClientRect();
+			const height = container ? `${container.width / this.aspectRatio}px` : '300px';
+			const width = container ? `${container.width}px` : '100%';
+			return {
+				height,
+				width,
+			};
+		},
+		refString() {
+			return `mapholder${this.mapId}`;
```

Comment:
I think the wrapperObserver might need to be destroyed if this changes, though this probably won't change in any of our use cases.

Reply:
We would want to refresh the observer if this changed, but probably also re-render the map...Shouldn't be an issue in the near term.